src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -102,16 +102,34 @@ CHIP_ERROR WifiSleepManager::VerifyAndTransitionToLowPowerMode()
 
     if (!(wifiConfig.ssid[0] != 0) && !isCommissioningInProgress)
     {
-        VerifyOrReturnError(wfx_power_save(RSI_SLEEP_MODE_8, DEEP_SLEEP_WITH_RAM_RETENTION, 0) != SL_STATUS_OK, CHIP_ERROR_INTERNAL,
-                            ChipLogError(DeviceLayer, "Failed to enable Deep Sleep."));
+        VerifyOrReturnError(CofigurePowerSave(RSI_SLEEP_MODE_8, DEEP_SLEEP_WITH_RAM_RETENTION, 0) != SL_STATUS_OK,
+                            CHIP_ERROR_INTERNAL, ChipLogError(DeviceLayer, "Failed to enable Deep Sleep."));
     }
     else
     {
-        VerifyOrReturnError(wfx_power_save(RSI_SLEEP_MODE_2, ASSOCIATED_POWER_SAVE, 0) != SL_STATUS_OK, CHIP_ERROR_INTERNAL,
-                            ChipLogError(DeviceLayer, "Failed to enable to go to sleep."));
+
+        if (mCallbacks && mCallbacks->CanGoToLIBasedSleep())
```

Comment:
How does that check work if, like the name suggests, there are more than one Callback?

Reply:
see other comment